sdk/protocol-manager-poc/src/implementation/sparkPlugin.ts

```diff
@@ -0,0 +1,167 @@
+import { AddressValue, Percentage, TokenAmount, TokenSymbol, Price, CurrencySymbol, RiskRatio } from "@summerfi/sdk-common/common"
+import type { SparkLendingPool, SparkPoolDebtConfig, SparkPoolCollateralConfig, SparkPoolId } from "@summerfi/sdk-common/protocols"
+import { PoolType, ProtocolName, EmodeType } from "@summerfi/sdk-common/protocols"
+import { Position, Token } from "@summerfi/sdk-common/common"
+import { BigNumber } from 'bignumber.js'
+import { z } from 'zod'
+import { ChainId, CreateProtocolPlugin, IPositionId, ProtocolManagerContext, ProtocolPlugin } from "../interfaces";
+import { AaveV3LikePluginBuilder, filterAssetsListByEMode } from "./aaveV3LikeBuilder";
+import { UNCAPPED_SUPPLY, PRESISION_BI } from "./constants";
+
+
+function tokenAmountFromBaseUnit({amount, token}: {amount: string, token: Token}): TokenAmount {
+    return TokenAmount.createFromBaseUnit({token, amount})
+}
+
+export const createSparkPlugin: CreateProtocolPlugin<SparkPoolId> = (ctx: ProtocolManagerContext): ProtocolPlugin<SparkPoolId> => {
+    const plugin: ProtocolPlugin<SparkPoolId> = {
+        supportedChains: [ChainId.Mainnet],
+        getPool: async (sparkPoolId: unknown): Promise<SparkLendingPool> => {
+            plugin._validate(sparkPoolId)
+            const emode = sparkEmodeCategoryMap[sparkPoolId.emodeType]
+            if (!emode && emode !== 0n) throw new Error('emode on poolId not recognised')
```

Comment:
could be moved to a validation fn as it checks the internal implementation details of sparkPoolId